pkg/planner/core/find_best_task.go

```diff
@@ -1129,38 +1129,50 @@ func compareCandidates(sctx base.PlanContext, statsTbl *statistics.Table, tableI
 		tablePseudo = statsTbl.HistColl.Pseudo
 		lhsPseudo, rhsPseudo = isCandidatesPseudo(lhs, rhs, lhsFullScan, rhsFullScan, statsTbl)
 	}
+	// matchResult: comparison result of whether LHS vs RHS matches the required properties (1=LHS better, -1=RHS better, 0=equal)
+	// globalResult: comparison result of global index vs local index preference (1=LHS better, -1=RHS better, 0=equal)
 	matchResult, globalResult := compareBool(lhs.isMatchProp, rhs.isMatchProp), compareGlobalIndex(lhs, rhs)
+	// accessResult: comparison result of access condition coverage (1=LHS better, -1=RHS better, 0=equal)
+	// comparable1: whether the access conditions are comparable between LHS and RHS
 	accessResult, comparable1 := util.CompareCol2Len(lhs.accessCondsColMap, rhs.accessCondsColMap)
+	// scanResult: comparison result of index back scan efficiency (1=LHS better, -1=RHS better, 0=equal)
+	// comparable2: whether the index back scan characteristics are comparable between LHS and RHS
 	scanResult, comparable2 := compareIndexBack(lhs, rhs)
+	// riskResult: comparison result of risk factor (1=LHS better, -1=RHS better, 0=equal)
 	riskResult, _ := compareRiskRatio(lhs, rhs)
-	sum := accessResult + scanResult + matchResult + globalResult
+	// eqOrInResult: comparison result of equal/IN predicate coverage (1=LHS better, -1=RHS better, 0=equal)
+	eqOrInResult, lhsEqOrInCount, rhsEqOrInCount := compareEqOrIn(lhs, rhs)
+
+	// sum is the aggregate score of various comparison metrics between lhs and rhs. riskResult is excluded
+	// because more work is required.
+	// TO-DO - extend riskResult such that risk factors can be integrated into the aggregate score. Risk should
```

Comment:
```suggestion
	// TODO: - extend riskResult such that risk factors can be integrated into the aggregate score. Risk should
```

We usually write it as `TODO:`, which allows some TODO management tools to automatically match, highlight, and remind developers.

<img width="1812" height="417" alt="image" src="https://github.com/user-attachments/assets/9fe0cca2-4913-406a-ad04-7fd163d7cda8" />